espresso/environment/3_2_espresso_deterministic_state_test.go

```diff
@@ -126,3 +209,133 @@ func TestDeterministicDerivationExecutionState(t *testing.T) {
+	if err != nil {
+		log.Error("failed to decode Espresso transaction in the test", "error", err)
+		return nil, err
+	}
+	buf := bytes.NewBuffer(bufData)
+
+	// Sign the encoded batch with FAKE or REAL batcher private key
+	batcherSignature, err := crypto.Sign(crypto.Keccak256(buf.Bytes()), batcherKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create batcher signature: %w", err)
+	}
+
+	// Combine signature and batch data
+	payload := append(batcherSignature, buf.Bytes()...)
+
+	// Create and return Espresso Transaction
+	return &espressoCommon.Transaction{
+		Namespace: chainID.Uint64(),
+		Payload:   payload,
+	}, nil
+}
+
+// TestValidEspressoTransactionCreation is a test that
+// make sure we have correct way to create a Espresso transaction.
+// This test is a unit test to serve the correctness of TestDeterministicDerivationExecutionStateWithInvalidTransaction.
+func TestValidEspressoTransactionCreation(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	launcher := new(env.EspressoDevNodeLauncherDocker)
+
+	// once this StartDevNet returns, we have a running Espresso Dev Node
+	system, espressoDevNode, err := launcher.StartDevNet(ctx, t, 0)
```

Comment:
```suggestion
	system, espressoDevNode, err := launcher.StartDevNet(ctx, t)
```

Reply:
The code does not compile otherwise.

Reply:
Btw, I don't think we should ever set it to `0` even if we do want to override the default value of the L1 finalized distance.